py-polars/polars/lazyframe/frame.py

```diff
@@ -615,7 +618,7 @@ def schema(self) -> SchemaDict:
         {'foo': Int64, 'bar': Float64, 'ham': Utf8}
 
         """
-        return self._ldf.schema()
```

Comment:
It would be better that this `_normalize_polars_dtype` would not be needed for dtypes and schema, but that the rust code for schema handles this directly:
https://github.com/stinodego/polars/blob/43c9def930a3c85511e02cbde02e525269f7e541/py-polars/src/lazy/dataframe.rs#L959-L969

Reply:
I agree, but this is not trivial. The datatypes are an Enum in Rust, and the actual code for converting to Python classes is here:
https://github.com/pola-rs/polars/blob/61ce6f44d04570f654ad3b0586bfdf15cda045d9/py-polars/src/conversion.rs#L707-L807

Currently, there are some Python utils used in the Rust bindings to create the complex datatypes. I'd have to introduce a util for the other types. Which I can do, but that is also kind of ugly.

I think it might be preferable to define the datatype classes as PyO3 classes. But I'd have to take some time to look into that more closely. For now I think the current solution is fine.

Reply:
This is the code for converting a python object to a rust anyvalue. The code for converting to python dtypes is much simpler.

Reply:
Hm, I may have been looking at the wrong thing then... let me look into this a bit more closely.